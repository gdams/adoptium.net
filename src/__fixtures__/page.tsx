import {
    AllReleaseNotes,
    AsciidocPage,
    File,
    ReleaseNotePage
} from '../types';

export const createAsciidocData = (): {
    asciidoc: AsciidocPage;
    file: File;
  } => ({
    asciidoc: {
        id: 'asciidoc-1',
        html: `
            <p>Hello World, this is a <pre><code>test</code></pre></p>
            <table><thead><td>Header 1</td><td>Header 2</td></thead><tbody><tr><td>Cell 1</td><td class="icon">Cell 2</td></tr></tbody></table>
            <a href="https://www.eclipse.org">Eclipse</a>
            <span class="icon">[docker]</span>
            <span class="icon">[check]</span>
        `,
        document: {
            title: 'Asciidoc Page title',
            main: 'Asciidoc Page content',
        },
        pageAttributes: {
            authors: 'author1, author2',
        },
        fields: {
            slug: '/asciidoc/asciidoc-page-title',
        }
    },
    file: {
        relativePath: 'test.adoc',
    }
});

export const createReleaseNoteData = (): {
    allReleaseNotes: AllReleaseNotes;
  } => ({
    allReleaseNotes: {
        edges: [
            {
                node: {
                    id: 'release-note-1',
                    version: '1.0.0',
                    title: 'Release Note title',
                    priority: 1,
                    component: 'component1',
                    link: 'https://release-note.com',
                }
            }
        ]
    }
});

