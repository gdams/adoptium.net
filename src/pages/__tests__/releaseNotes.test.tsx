import React from 'react';
import { render } from '@testing-library/react';
import AllReleaseNotesPages from '../../templates/releaseNotesTemplate';
import { describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe';
import { createReleaseNoteData } from '../../__fixtures__/page';

let mockData = createReleaseNoteData();

describe('ReleaseNote pages', () => {
  it('renders correctly', () => {
    const { container } = render(<AllReleaseNotesPages pageContext={{version: '1.0.0'}} data={mockData} />);
    // eslint-disable-next-line
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });

  it('has no accessibility violations', async () => {
    const { container } = render(<AllReleaseNotesPages pageContext={{version: '1.0.0'}} data={mockData} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});