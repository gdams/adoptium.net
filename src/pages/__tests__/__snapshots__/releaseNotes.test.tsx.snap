// Vitest Snapshot v1

exports[`ReleaseNote pages > renders correctly 1`] = `
<main>
  <title>
    Release Notes - 1.0.0 | Adoptium
  </title>
  <meta
    content="Sample Description"
    name="description"
  />
  <meta
    content="Release Notes - 1.0.0 | Adoptium"
    name="og:title"
  />
  <meta
    content="Sample Description"
    name="og:description"
  />
  <meta
    content="website"
    name="og:type"
  />
  <meta
    content="https://adoptium.net/images/social-image.png"
    name="og:image"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@adoptium"
    name="twitter:site"
  />
  <meta
    content="https://adoptium.net/images/social-image.png"
    name="twitter:image"
  />
  <meta
    content="@adoptium"
    name="twitter:creator"
  />
  <meta
    content="Release Notes - 1.0.0 | Adoptium"
    name="twitter:title"
  />
  <meta
    content="Sample Description"
    name="twitter:description"
  />
  <link
    href="//www.eclipse.org/eclipse.org-common/themes/solstice/public/stylesheets/vendor/cookieconsent/cookieconsent.min.css"
    rel="stylesheet"
    type="text/css"
  />
  <script
    src="//www.eclipse.org/eclipse.org-common/themes/solstice/public/javascript/vendor/cookieconsent/default.min.js"
  />
  <section
    class="py-5 container-xxl"
  >
    <h1
      class="fw-light text-center"
    >
      Text
       - 1.0.0
    </h1>
    <div
      class="p-5"
    >
      <label
        class="search-label"
        for="search-bar-0"
      >
        <span
          class="sr-only"
          id="search-bar-0-label"
        >
          Search this table
        </span>
        <input
          aria-labelledby="search-bar-0-label"
          class="form-control "
          id="search-bar-0"
          placeholder="Search"
          type="text"
          value=""
        />
      </label>
      <div
        class="react-bootstrap-table"
      >
        <table
          class="table table-striped table-hover table-bordered"
        >
          <thead>
            <tr>
              <th
                aria-label="Issue sortable"
                class="sortable"
                tabindex="0"
              >
                Issue
                <span
                  class="order"
                >
                  <span
                    class="dropdown"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                  <span
                    class="dropup"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                </span>
              </th>
              <th
                aria-label="Component sortable"
                class="sortable"
                tabindex="0"
              >
                Component
                <span
                  class="order"
                >
                  <span
                    class="dropdown"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                  <span
                    class="dropup"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                </span>
              </th>
              <th
                aria-label="Priority sortable"
                class="sortable"
                tabindex="0"
              >
                Priority
                <span
                  class="order"
                >
                  <span
                    class="dropdown"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                  <span
                    class="dropup"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                </span>
              </th>
              <th
                tabindex="0"
              >
                Title
              </th>
            </tr>
          </thead>
          <tbody
            class="text-nowrap"
          >
            <tr>
              <td>
                <a
                  href="https://release-note.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  release-note-1
                </a>
              </td>
              <td />
              <td>
                1
              </td>
              <td>
                Release Note title
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="row react-bootstrap-table-pagination"
      >
        <div
          class="col-md-6 col-xs-6 col-sm-6 col-lg-6"
        >
          <span
            class="react-bs-table-sizePerPage-dropdown dropdown"
            style="visibility: visible;"
          >
            <button
              aria-expanded="false"
              class="btn btn-default btn-secondary dropdown-toggle"
              data-toggle="dropdown"
              id="pageDropDown"
              type="button"
            >
              20
               
              <span>
                <span
                  class="caret"
                />
              </span>
            </button>
            <ul
              aria-labelledby="pageDropDown"
              class="dropdown-menu "
              role="menu"
            >
              <li
                class="dropdown-item"
                role="presentation"
              >
                <a
                  data-page="10"
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  10
                </a>
              </li>
              <li
                class="dropdown-item"
                role="presentation"
              >
                <a
                  data-page="25"
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  25
                </a>
              </li>
              <li
                class="dropdown-item"
                role="presentation"
              >
                <a
                  data-page="30"
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  30
                </a>
              </li>
              <li
                class="dropdown-item"
                role="presentation"
              >
                <a
                  data-page="50"
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  50
                </a>
              </li>
            </ul>
          </span>
          <span
            class="react-bootstrap-table-pagination-total"
          >
             Showing rows 
            1
             to 
            1
             of 
            1
          </span>
        </div>
        <div
          class="react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6"
        >
          <ul
            class="pagination react-bootstrap-table-page-btns-ul"
          >
            <li
              class="active page-item"
              title="1"
            >
              <a
                class="page-link"
                href="#"
              >
                1
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </section>
</main>
`;
